db/model/collector.js

```diff
@@ -253,7 +242,10 @@ module.exports = function collector(seq, dataTypes) {
   Collector.prototype.reassignGenerators = function () {
     /* TODO: change to use currentGenerators once that includes current gens only */
     return seq.models.Generator.findAll({ where: { currentCollector: this.name } })
-    .then((gens) => Promise.all(gens.map((g) => g.assignToCollector())));
+    .map((g) => {
```

Comment:
is the TODO on line 243 still valid?

Reply:
Yes. Once @sameesiddiqui 's change is done we can replace the findAll with this.currentGenerators